packages/composites/the-graph/src/methods/prices/prices.ts

```
@@ -16,111 +19,161 @@ const customParams = {
-  }
-  const inputParams: DexQueryInputParams = {
-    jobRunID,
-    baseCoinTicker: baseCoinTicker.toUpperCase(),
-    quoteCoinTicker: theGraphQuote ? theGraphQuote.toUpperCase() : quoteCoinTicker.toUpperCase(),
-    dex: dexToUpperCase,
-    referenceContract,
-    referenceContractDivisor,
-    referenceModifierAction: referenceModifierAction.toUpperCase() as ReferenceModifierAction,
-    intermediaryToken: intermediaryToken.toUpperCase(),
-  }
-  if (baseCoinTicker === quoteCoinTicker) {
-    throw new Error('Base and Quote coins must be different')
-  }
-  Logger.info(`Fetching quote for ${quoteCoinTicker}/${baseCoinTicker} pair from ${dex}`)
-  let price
-  try {
-    price = await getQuotePrice(inputParams, dexSubgraph)
-  } catch (e) {
-    throw new Error(`Failed to get price.  Reason "${e}"`)
-  }
-  return Requester.success(
-    jobRunID,
-    {
-      status: 200,
-      data: {
-        result: price,
-      },
-    },
-    true,
-  )
+export const execute = async (input: AdapterRequest, config: Config): Promise<AdapterResponse> => {
+    const validator = new Validator(input, customParams)
+    if (validator.error) throw validator.error
+    const jobRunID = validator.validated.id
+    const inputParams = validator.validated.data
+    if (!inputParams.theGraphQuote && !inputParams.quoteCoinTicker) {
+        throw new Error("quoteCoinTicker cannot be empty if theGraphQuote not supplied")
+    }
+    const dexToUpperCase = inputParams.dex.toUpperCase()
+    const dexSubgraph = config.dexSubgraphs[dexToUpperCase]
+    if (!dexSubgraph) {
+        throw new Error(`${inputParams.dex} is currently not supported`)
+    }
+    let price
+    switch(dexToUpperCase) {
+        case UNISWAP:
+            price = await executeUniswap(jobRunID, inputParams, dexSubgraph as UniswapSubgraph)
+            break;
+        case CURVE:
+            price = await executeCurve(jobRunID, inputParams, dexSubgraph as CurveSubgraph)
+            break;
+        default:
```

We should throw an error instead as we don't want a default dex.